.github/workflows/Trgr InitDev workflows in Implementation repo.yml

```diff
@@ -57,8 +57,44 @@ jobs:
           $current_branch = '${{env.HEAD_REF}}'
         }
         Write-Host "Current Branch: $current_branch"   
-        echo "current_branch=$current_branch" >> $Env:GITHUB_ENV                   
+        echo "current_branch=$current_branch" >> $Env:GITHUB_ENV
+    - name: Check Ed-Fi-ODS-Implementation Repo PR exists or not
+      shell: pwsh
+      run: |
+        $accessToken = "${{ env.EDFI_ODS_IMP_TOKEN }}"
+        $repositoryName = "Ed-Fi-ODS-Implementation"
+        $apiUrl = "https://api.github.com/repos/${{ env.REPOSITORY_OWNER }}/$repositoryName/pulls?state=open&head=${{ env.REPOSITORY_OWNER }}:${{ env.current_branch }}"
+        $headers = @{
+            Authorization = "Bearer $accessToken"
+            Accept = "application/vnd.github.v3+json"
+        }
+        $response = Invoke-WebRequest -Uri $apiUrl -Headers $headers
+
+        # Access x-ratelimit-limit and x-ratelimit-remaining from response headers
+        $rateLimitLimit = $response.Headers['X-RateLimit-Limit']
+        $rateLimitRemaining = $response.Headers['X-RateLimit-Remaining']
+
+
+        Write-Host "x-ratelimit-limit: $rateLimitLimit"
+        Write-Host "x-ratelimit-remaining: $rateLimitRemaining"
+
+        $jsonResponse = $response | ConvertFrom-Json
+
+        foreach ($pr in $jsonResponse) {
+            Write-Host "current branch: ${{ env.current_branch }}"
+            Write-Host "ref: $($pr.head.ref)"
+            if ($pr.head.ref -eq "${{ env.current_branch }}") 
+            {
+                Write-Host "PR #$($pr.number): $($pr.head.label)"
+                echo "EXIT_STEP=true">> $env:GITHUB_ENV
+            }
+        }
+    - name: Check for EXIT_STEP Status
+      if: env.EXIT_STEP == 'true'
+      run: | 
+           echo "::notice::Ed-Fi-ODS-Implementation already have same branch ${{ env.current_branch }} running all Initdev builds ,so Skipping this build"
```

Comment:
Oh, I completely misunderstood the intent of this pull request before! But now I get it after reading note. I still find it strange that you have used `echo` here instead of `Write-Host` as seen elsewhere. This is good and makes a lot of sense.

Another thing I would like to see is in the remote job in `Ed-Fi-ODS-Implementation` is that the job quits if the repo doesn't have the required branch, instead of retrying over and over. We can do that in another ticket.

Reply:
why I used echo for setting environment variable in PowerShell  echo "EXIT_STEP=true">> $env:GITHUB_ENV https://www.jamescroft.co.uk/setting-github-actions-environment-variables-in-powershell/

Reply:
And also I used in many other builds as well

Reply:
In PowerShell, `echo` is just an alias of [Write-Output](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-output?view=powershell-7.3). Also, `Write-Host` and `Write-Output` are very similar; after many years of using them, I still haven't fully understood why they both exist. I'm not going to stop the PR over this topic, but our official coding standard for PowerShell is to write out the commands in their full PowerShell form, not in shortcut form like `echo` (another example: using `Remove-Item` instead of `rm`).